### PR TITLE
The Epic Construction Update

### PR DESCRIPTION
**Pull request by LadyFowls**
https://github.com/Aurorastation/Aurora.3/pull/5960

How to build machine blueprints!

Use steel sheets like normal, then rotate the frame how you like it using the directional arrow. From here use a Multitool to finalize it and then wire it up like you would before.

### CODE REVIEW

File: code/game/machinery/constructable_frame.dm [hunk 1]
```diff
@@ -3,61 +3,89 @@
...
+				component_list += "[num2text(req_components[I])] [req_component_names[I]]"
+		D = "Requires [english_list(component_list)]."
+	desc = D
 
 /obj/machinery/constructable_frame/machine_frame
 	attackby(obj/item/P as obj, mob/user as mob)
 		switch(state)
 			if(1)
+				if(P.ispen())
+					to_chat(user, span("notice", "You begin to finalize the blueprint.</span>"))
```
Comment (MarinaGryphon) [inline code/game/machinery/constructable_frame.dm]: Remove the `</span>` here.

File: code/game/machinery/constructable_frame.dm [hunk 2]
```diff
@@ -3,61 +3,89 @@
...
+					if(do_after(user, 20) && state == 1)
+						to_chat(user, span("notice", "You finalize the blueprint.</span>"))
```
Comment (MarinaGryphon) [inline code/game/machinery/constructable_frame.dm]: Remove the `</span>` here.

File: code/game/machinery/constructable_frame.dm [hunk 3]
```diff
@@ -3,61 +3,89 @@
...
+						playsound(src.loc, 'sound/items/poster_being_created.ogg', 75, 1)
+						state = 2
+				else
+					if(P.ismultitool())
+						playsound(src.loc, 'sound/items/poster_ripped.ogg', 75, 1)
+						to_chat(user, span("notice", "You decide to scrap the blueprint</span>"))
```
Comment (MarinaGryphon) [inline code/game/machinery/constructable_frame.dm]: Remove the `</span>` here.

File: code/game/machinery/constructable_frame.dm [hunk 4]
```diff
@@ -3,61 +3,89 @@
...
 				if(P.iscoil())
 					var/obj/item/stack/cable_coil/C = P
 					if (C.get_amount() < 5)
-						user << "<span class='warning'>You need five lengths of cable to add them to the frame.</span>"
+						to_chat(user, span("notice", "You need five lengths of cable to add them to the blueprint.</span>"))
 						return
 					playsound(src.loc, 'sound/items/Deconstruct.ogg', 50, 1)
-					user << "<span class='notice'>You start to add cables to the frame.</span>"
-					if(do_after(user, 20) && state == 1)
+					to_chat(user, span("notice", "You start to add cables to the blueprint.</span>"))
```
Comment (MarinaGryphon) [inline code/game/machinery/constructable_frame.dm]: Remove the `</span>` here.

File: code/game/machinery/constructable_frame.dm [hunk 5]
```diff
@@ -3,61 +3,89 @@
...
-							state = 2
-							icon_state = "box_1"
-				else
-					if(P.iswrench())
-						playsound(src.loc, 'sound/items/Ratchet.ogg', 75, 1)
-						user << "<span class='notice'>You dismantle the frame</span>"
-						new /obj/item/stack/material/steel(src.loc, 5)
-						qdel(src)
-			if(2)
+							to_chat(user, span("notice", "You add cables to the blueprint.</span>"))
```
Comment (MarinaGryphon) [inline code/game/machinery/constructable_frame.dm]: Remove the `</span>` here.

File: code/game/machinery/constructable_frame.dm [hunk 6]
```diff
@@ -3,61 +3,89 @@
...
+							state = 3
+							icon_state = "blueprint_2"
+			if(3)
 				if(istype(P, /obj/item/weapon/circuitboard))
 					var/obj/item/weapon/circuitboard/B = P
 					if(B.board_type == "machine")
 						playsound(src.loc, 'sound/items/Deconstruct.ogg', 50, 1)
-						user << "<span class='notice'>You add the circuit board to the frame.</span>"
+						to_chat(user, span("notice", "You add the circuit board to the blueprint.</span>"))
```
Comment (MarinaGryphon) [inline code/game/machinery/constructable_frame.dm]: Remove the `</span>` here.

**Comment by dragoslava** [comment]
Gonna start making a changelog as this is getting bigger

File: code/game/machinery/constructable_frame.dm [hunk 7]
```diff
@@ -3,61 +3,89 @@
...
-				if(req_components[I] > 0)
-					component_list += "[num2text(req_components[I])] [req_component_names[I]]"
-			D = "Requires [english_list(component_list)]."
-		desc = D
+/obj/machinery/constructable_frame/verb/rotate_clockwise()
+	set category = "Object"
+	set name = "Rotate (Clockwise)"
+	set src in view(1)
+
+	if (usr.stat || usr.restrained() || anchored)
```
[the remark this answers is not in the document]
Comment (MarinaGryphon) [reply code/game/machinery/constructable_frame.dm]: Keep in mind that you want to check for if `use_check(usr)` returns TRUE. If it returns true, the user CANNOT use it.

File: code/game/machinery/constructable_frame.dm [hunk 8]
```diff
@@ -3,61 +3,89 @@
...
+
+	src.set_dir(turn(src.dir, -90))
+
+
+/obj/machinery/constructable_frame/verb/rotate_counterclockwise()
+	set category = "Object"
+	set name = "Rotate (Counterclockwise)"
+	set src in view(1)
+
+	if (usr.stat || usr.restrained() || anchored)
```
[the remark this answers is not in the document]
Comment (MarinaGryphon) [reply code/game/machinery/constructable_frame.dm]: Keep in mind that you want to check for if `use_check(usr)` returns TRUE. If it returns true, the user CANNOT use it.

File: code/game/machinery/constructable_frame.dm [hunk 9]
```diff
@@ -3,61 +3,89 @@
...
+			if(req_components[I] > 0)
+				component_list += "[num2text(req_components[I])] [req_component_names[I]]"
+		D = "Requires [english_list(component_list)]."
+	desc = D
 
 /obj/machinery/constructable_frame/machine_frame
 	attackby(obj/item/P as obj, mob/user as mob)
 		switch(state)
 			if(1)
+				if(P.ispen())
```
[the remark this answers is not in the document]
Comment (MarinaGryphon) [reply code/game/machinery/constructable_frame.dm]: We have pens in PDAs and it avoids conflicts between other tools. I think the idea was that it's a blueprint that you're modifying?
Even if it's a holographic blueprint, it reminds me of those "Promethean boards" or whatever and how they had special pens for them that teachers would use to write stuff on screen/control their computer.

File: vueui/dist/app.js [hunk 10]
```diff
@@ -1,8 +1,8 @@
...
...atch(r){}}(),k=function(t){var n;return!(!p(t)||"function"!=typeof(n=t.then))&&n},N=function(t,n){if(!t._n){t._n=!0;var e=t._c;g(function(){var r=t._v,o=1==t._s,i=0,a=function(n){var e,i,a,s=o?n.ok:n.fail,u=n.resolve,c=n.reject,f=n.domain;try{s?(o||(2==t._h&&P(t),t._h=1),!0===s?e=r:(f&&f.enter(),e=s(r),f&&(f.exit(),a=!0)),e===n.promise?c(S("Promise-chain cycle")):(i=k(e))?i.call(e,u,c):u(e)):c(r)}catch(l){f&&!a&&f.exit(),c(l)}};while(e.length>i)a(e[i++]);t._c=[],t._n=!1,n&&!t._h&&F(t)})}},F=function(t){_.call(u,function(){var n,e,r,o=t._v,i=M(t);if(i&&(n=m(function(){E?$.emit("unhandledRejection",o,t):(e=u.onunhandledrejection)?e({promise:t,reason:o}):(r=u.console)&&r.error&&r.error("Unhandled promise rejection",o)}),t._h=E||M(t)?2:1),t._a=void 0,i&&n.e)throw n.v})},M=function(t){return 1!==t._h&&0===(t._a||t._c).length},P=function(t){_.call(u,function(){var n;E?$.emit("rejectionHandled",t):(n=u.onrejectionhandled)&&n({promise:t,reason:t._v})})},U=function(t){var n=this;n._d||(n._d=!0,n=n._w||n,n._v=t,n._s=2,n._a||(n._a=n._c.slice()),N(n,!0))},R=function(t){var n,e=this;if(!e._d){e._d=!0,e=e._w||e;try{if(e===t)throw S("Promise can't be resolved itself");(n=k(t))?g(function(){var r=
```
Comment (avabee) [inline vueui/dist/app.js]: You need to revert these changes.

**Comment by skull132** [comment]
image
this plz fix.

**Comment by dragoslava** [comment]
> image
> this plz fix.

https://i.imgur.com/zYiFja1.png

**Comment by Sindorman** [comment]
All new UIs needs to be using VueUi

File: code/ATMOSPHERICS/components/unary/vent_pump.dm [hunk 11]
```diff
@@ -400,9 +400,9 @@
 		to_chat(user, "<span class='warning'>You cannot unwrench \the [src], it is too exerted due to internal pressure.</span>")
 		add_fingerprint(user)
 		return 1
-	playsound(src.loc, 'sound/items/Ratchet.ogg', 50, 1)
+	playsound(src.loc, W.usesound, 50, 1)
 	to_chat(user, "<span class='notice'>You begin to unfasten \the [src]...</span>")
-	if (do_after(user, 40))
+	if (do_after(user, 40)/W.toolspeed)
```
Comment (SoundScopes) [inline code/ATMOSPHERICS/components/unary/vent_pump.dm]: "/W.toolspeed" isn't in the do_after argument

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 12]
```diff
@@ -276,7 +276,7 @@
 				"You hear a welding torch on metal."
 			)
 			playsound(loc, 'sound/items/Welder2.ogg', 50, 1)
-			if (!do_after(user, 2 SECONDS, act_target = src, extra_checks = CALLBACK(src, .proc/is_closed)))
+			if (!do_after(user, 2/W.toolspeed, act_target = src, extra_checks = CALLBACK(src, .proc/is_closed)))
```
Comment (SoundScopes) [inline code/game/objects/structures/crates_lockers/closets.dm]: This change will make it happen after 2 ticks / toolspeed instead of 2 seconds / toolspeed

File: code/game/objects/structures/crates_lockers/closets/secure/secure_closets.dm [hunk 13]
```diff
@@ -87,7 +87,7 @@
 					"You hear a welding torch on metal."
 				)
 				playsound(loc, 'sound/items/Welder2.ogg', 50, 1)
-				if (!do_after(user, 2 SECONDS, act_target = src, extra_checks = CALLBACK(src, .proc/is_open)))
+				if (!do_after(user, 2/W.toolspeed, act_target = src, extra_checks = CALLBACK(src, .proc/is_open)))
```
Comment (SoundScopes) [inline code/game/objects/structures/crates_lockers/closets/secure/secure_closets.dm]: This change will make it happen after 2 ticks / toolspeed instead of 2 seconds / toolspeed

File: code/game/objects/structures/crates_lockers/closets/secure/secure_closets.dm [hunk 14]
```diff
@@ -111,33 +111,33 @@
 	else if(W.isscrewdriver() && canbemoved)
 		if(screwed)
 			to_chat(user,  "<span class='notice'>You start to unscrew the locker from the floor...</span>")
-			playsound(loc, 'sound/items/Screwdriver.ogg', 50, 1)
-			if (do_after(user, 10 SECONDS, act_target = src))
+			playsound(loc, W.usesound, 50, 1)
+			if (do_after(user, 10/W.toolspeed, act_target = src))
```
Comment (SoundScopes) [inline code/game/objects/structures/crates_lockers/closets/secure/secure_closets.dm]: This change will make it happen after 10 ticks / toolspeed instead of 10 seconds / toolspeed

File: code/game/objects/structures/crates_lockers/closets/secure/secure_closets.dm [hunk 15]
```diff
@@ -111,33 +111,33 @@
...
-				playsound(loc, 'sound/items/Ratchet.ogg', 50, 1)
+				playsound(loc, W.usesound, 50, 1)
 				wrenched = 0
 				anchored = 0
 		else if(!wrenched)
 			to_chat(user,  "<span class='notice'>You start to fasten the bolts holding the locker in place...</span>")
-			playsound(loc, 'sound/items/Ratchet.ogg', 50, 1)
-			if (do_after(user, 15, act_target = src))
+			playsound(loc, W.usesound, 50, 1)
+			if (do_after(user, 15/W.toolspeed, act_target = src))
```
Comment (SoundScopes) [inline code/game/objects/structures/crates_lockers/closets/secure/secure_closets.dm]: Same as above, 15 ticks / toolspeed instead of 15 seconds / toolspeed

File: code/game/objects/structures/crates_lockers/closets/secure/secure_closets.dm [hunk 16]
```diff
@@ -111,33 +111,33 @@
...
-				playsound(loc, 'sound/items/Screwdriver.ogg', 50, 1)
+				playsound(loc, W.usesound, 50, 1)
 				screwed = 1
 	else if(W.iswrench() && canbemoved)
 		if(wrenched && !screwed)
 			to_chat(user,  "<span class='notice'>You start to unfasten the bolts holding the locker in place...</span>")
-			playsound(loc, 'sound/items/Ratchet.ogg', 50, 1)
-			if (do_after(user, 15 SECONDS, act_target = src))
+			playsound(loc, W.usesound, 50, 1)
+			if (do_after(user, 15/W.toolspeed, act_target = src))
```
Comment (SoundScopes) [inline code/game/objects/structures/crates_lockers/closets/secure/secure_closets.dm]: Same as above, 15 ticks / toolspeed instead of 15 seconds / toolspeed

File: code/game/turfs/simulated/wall_attacks.dm [hunk 17]
```diff
@@ -228,7 +228,7 @@
 			if(cut_delay<0)
 				cut_delay = 0
 
-			if(!do_after(user,cut_delay))
+			if(!do_after(user,4/W.toolspeed))
```
Comment (SoundScopes) [inline code/game/turfs/simulated/wall_attacks.dm]: Each material has it's own delay depending on how strong it is, this should be "cut_delay/W.toolspeed"
and the above line should be set to 1 to avoid dividing by 0

File: code/modules/power/apc.dm [hunk 18]
```diff
@@ -658,16 +658,16 @@
 	else if (istype(W, /obj/item/device/debugger))
 		if(emagged || hacker || infected)
 			to_chat(user, "<span class='warning'>There is a software error with the device. Attempting to fix...</span>")
-			if(do_after(user, 10 SECONDS, act_target = src))
+			if(do_after(user, 10/W.toolspeed SECONDS, act_target = src))
```
Comment (SoundScopes) [inline code/modules/power/apc.dm]: Lines 661, 663, 665, instead of "X seconds / toolspeed" have changed to "X ticks / toolspeed"

Comment (SoundScopes) [reply code/modules/power/apc.dm]: Apologies I appear to be blind on this one

File: code/game/turfs/simulated/wall_attacks.dm [hunk 19]
```diff
@@ -228,7 +228,7 @@
 			if(cut_delay<0)
 				cut_delay = 0
 
-			if(!do_after(user,cut_delay))
+			if(!do_after(user,cut_delay/W.toolspeed))
```
Comment (SoundScopes) [inline code/game/turfs/simulated/wall_attacks.dm]: If cut_delay is <0  then it gets set to 0, then it will divide by 0 and runtime

File: code/game/machinery/computer/Operating.dm [hunk 20]
```diff
@@ -82,3 +82,164 @@
 /obj/machinery/computer/operating/machinery_process()
 	if(operable())
 		src.updateDialog()
+
+
+
+
+/obj/machinery/computer/operating/advanced
+	name = "Advanced Operating Computer"
```
Comment (NonQueueingMatt) [inline code/game/machinery/computer/Operating.dm]:
```suggestion
	name = "advanced operating computer"
```

File: code/game/machinery/constructable_frame.dm [hunk 21]
```diff
@@ -3,61 +3,96 @@
...
-		if(req_components)
-			var/list/component_list = new
-			for(var/I in req_components)
-				if(req_components[I] > 0)
-					component_list += "[num2text(req_components[I])] [req_component_names[I]]"
-			D = "Requires [english_list(component_list)]."
-		desc = D
+/obj/machinery/constructable_frame/verb/rotate_clockwise()
+	set category = "Object"
+	set name = "Rotate (Clockwise)"
```
Comment (NonQueueingMatt) [inline code/game/machinery/constructable_frame.dm]:
```suggestion
	set name = "Rotate Clockwise"
```

File: code/game/machinery/constructable_frame.dm [hunk 22]
```diff
@@ -3,61 +3,96 @@
...
+
+	if(use_check(usr) || anchored)
+		return
+
+	src.set_dir(turn(src.dir, -90))
+
+
+/obj/machinery/constructable_frame/verb/rotate_counterclockwise()
+	set category = "Object"
+	set name = "Rotate (Counterclockwise)"
```
Comment (NonQueueingMatt) [inline code/game/machinery/constructable_frame.dm]:
```suggestion
	set name = "Rotate Counter-Clockwise"
```

File: code/game/machinery/constructable_frame.dm [hunk 23]
```diff
@@ -3,61 +3,96 @@
...
+				if(P.ismultitool())
+					to_chat(user, span("notice", "You begin to finalize the blueprint."))
+					if(do_after(user, 20) && state == 1)
+						to_chat(user, span("notice", "You finalize the blueprint."))
+						playsound(src.loc, 'sound/items/poster_being_created.ogg', 75, 1)
+						state = 2
+				else
+					if(P.iswirecutter())
+						playsound(src.loc, 'sound/items/poster_ripped.ogg', 75, 1)
+						to_chat(user, span("notice", "You decide to scrap the blueprint"))
```
Comment (NonQueueingMatt) [inline code/game/machinery/constructable_frame.dm]:
```suggestion
						to_chat(user, span("notice", "You decide to scrap the blueprint."))
```

File: code/game/machinery/kitchen/smartfridge.dm [hunk 24]
```diff
@@ -250,9 +258,41 @@
 
 	if(O.ismultitool()||O.iswirecutter())
 		if(panel_open)
+			switch(input(user, "What would you like to select?", "Machine Debug Software") as null|anything in list("SmartHeater", "MegaSeed Storage", "Slime Extract Storage", "Refrigerated Chemical Storage", "Refrigerated Virus Storage", "Drink Showcase", "Drying Rack"))
+				if("SmartHeater.")
+					new /obj/machinery/smartfridge/foodheater(loc, make_from=src)
+					qdel(src)
+
+				if("MegaSeed Storage.")
```
Comment (NonQueueingMatt) [inline code/game/machinery/kitchen/smartfridge.dm]:
```suggestion
				if("MegaSeed Storage")
```

File: code/game/machinery/kitchen/smartfridge.dm [hunk 25]
```diff
@@ -250,9 +258,41 @@
 
 	if(O.ismultitool()||O.iswirecutter())
 		if(panel_open)
+			switch(input(user, "What would you like to select?", "Machine Debug Software") as null|anything in list("SmartHeater", "MegaSeed Storage", "Slime Extract Storage", "Refrigerated Chemical Storage", "Refrigerated Virus Storage", "Drink Showcase", "Drying Rack"))
+				if("SmartHeater.")
```
Comment (NonQueueingMatt) [inline code/game/machinery/kitchen/smartfridge.dm]:
```suggestion
				if("SmartHeater.")
```
```suggestion
				if("SmartHeater")
```

File: code/game/machinery/kitchen/smartfridge.dm [hunk 26]
```diff
@@ -250,9 +258,41 @@
...
+			switch(input(user, "What would you like to select?", "Machine Debug Software") as null|anything in list("SmartHeater", "MegaSeed Storage", "Slime Extract Storage", "Refrigerated Chemical Storage", "Refrigerated Virus Storage", "Drink Showcase", "Drying Rack"))
+				if("SmartHeater.")
+					new /obj/machinery/smartfridge/foodheater(loc, make_from=src)
+					qdel(src)
+
+				if("MegaSeed Storage.")
+					new /obj/machinery/smartfridge/seeds(loc, make_from=src)
+					qdel(src)
+
+				if("Slime Extract Storage.")
```
Comment (NonQueueingMatt) [inline code/game/machinery/kitchen/smartfridge.dm]:
```suggestion
				if("Slime Extract Storage")
```

File: code/game/machinery/kitchen/smartfridge.dm [hunk 27]
```diff
@@ -250,9 +258,41 @@
...
+					new /obj/machinery/smartfridge/secure/extract(loc, make_from=src)
+					qdel(src)
+
+				if("Refrigerated Chemical Storage.")
```
Comment (NonQueueingMatt) [inline code/game/machinery/kitchen/smartfridge.dm]:
```suggestion
				if("Refrigerated Chemical Storage")
```

File: code/game/machinery/kitchen/smartfridge.dm [hunk 28]
```diff
@@ -250,9 +258,41 @@
...
+					new /obj/machinery/smartfridge/secure/medbay(loc, make_from=src)
+					qdel(src)
+
+				if("Refrigerated Virus Storage.")
```
Comment (NonQueueingMatt) [inline code/game/machinery/kitchen/smartfridge.dm]:
```suggestion
				if("Refrigerated Virus Storage")
```

File: code/game/machinery/kitchen/smartfridge.dm [hunk 29]
```diff
@@ -250,9 +258,41 @@
...
+					new /obj/machinery/smartfridge/secure/virology(loc, make_from=src)
+					qdel(src)
+
+				if("Drink Showcase.")
```
Comment (NonQueueingMatt) [inline code/game/machinery/kitchen/smartfridge.dm]:
```suggestion
				if("Drink Showcase")
```

File: code/game/machinery/kitchen/smartfridge.dm [hunk 30]
```diff
@@ -250,9 +258,41 @@
...
+					new /obj/machinery/smartfridge/drinks(loc, make_from=src)
+					qdel(src)
+
+				if("Drying Rack.")
```
Comment (NonQueueingMatt) [inline code/game/machinery/kitchen/smartfridge.dm]:
```suggestion
				if("Drying Rack")
```

File: code/game/machinery/optableadv.dm [hunk 31]
```diff
@@ -0,0 +1,454 @@
...
+#define AUTO_AIR 2
+#define TRANSFUSE_ACTIVE 4
+#define AIR_ACTIVE 8
+#define PURGE 16
+
+/*
+ * The more advanced life support table
+ */
+/obj/machinery/optable/lifesupport
+	name = "Life Support Enabled Operating Table"
```
Comment (NonQueueingMatt) [inline code/game/machinery/optableadv.dm]:
```suggestion
	name = "life support operating table"
```

File: code/game/machinery/optableadv.dm [hunk 32]
```diff
@@ -0,0 +1,454 @@
...
+
+	if (istype(W, /obj/item/weapon/reagent_containers/blood))
+		if (!bloodbag)
+			user.drop_item()
+			bloodbag = W
+			W.loc = src
+			user << "<span class='notice'>You load [W] into [src], and hook it up to the machine.</span>"
+			return
+		else
+			user << "<span class='notice'>There is already a bloodbag in [src]!</span>"
```
Comment (NonQueueingMatt) [inline code/game/machinery/optableadv.dm]:
```suggestion
			to_chat(user, "<span class='notice'>There is already a bloodbag in [src]!</span>")
```

File: code/game/machinery/optableadv.dm [hunk 33]
```diff
@@ -0,0 +1,454 @@
...
+	. = ..()
+	if (.)
+		return
+
+	if (istype(W, /obj/item/weapon/reagent_containers/blood))
+		if (!bloodbag)
+			user.drop_item()
+			bloodbag = W
+			W.loc = src
+			user << "<span class='notice'>You load [W] into [src], and hook it up to the machine.</span>"
```
Comment (NonQueueingMatt) [inline code/game/machinery/optableadv.dm]:
```suggestion
			to_chat(user, "<span class='notice'>You load [W] into [src], and hook it up to the machine.</span>")
```

File: code/game/machinery/optableadv.dm [hunk 34]
```diff
@@ -0,0 +1,454 @@
...
+
+	if (W.is_open_container())
+		return 0
+
+	if (istype(W, /obj/item/weapon/tank))
+		if (!airsupply)
+			user.drop_item()
+			airsupply = W
+			W.loc = src
+			user << "<span class='notice'>You load [W] into [src], and hook it up the machine.</span>"
```
Comment (NonQueueingMatt) [inline code/game/machinery/optableadv.dm]:
```suggestion
			to_chat(user, "<span class='notice'>You load [W] into [src], and hook it up the machine.</span>")
```

File: code/game/machinery/optableadv.dm [hunk 35]
```diff
@@ -0,0 +1,454 @@
...
+			return
+		else
+			user << "<span class='notice'>There is already [airsupply] in [src]!</span>"
```
Comment (NonQueueingMatt) [inline code/game/machinery/optableadv.dm]:
```suggestion
			to_chat(user, "<span class='notice'>There is already [airsupply] in [src]!</span>")
```

File: code/game/machinery/optableadv.dm [hunk 36]
```diff
@@ -0,0 +1,454 @@
...
+			return
+
+	if (istype(W, /obj/item/weapon/card/emag))
+		if (!emagged)
+			emagged = 1
+			user << "<span class='notice'>You run [W] through [src], hear the machine quietly whirr. A new option has been unlocked.</span>"
+			return
```
Comment (NonQueueingMatt) [inline code/game/machinery/optableadv.dm]: to_chat

File: code/game/machinery/optableadv.dm [hunk 37]
```diff
@@ -0,0 +1,454 @@
...
+		else
+			user << "<span class='notice'>[src] is already emagged!</span>"
```
Comment (NonQueueingMatt) [inline code/game/machinery/optableadv.dm]: to_chat

File: code/game/machinery/optableadv.dm [hunk 38]
```diff
@@ -0,0 +1,454 @@
...
+
+		if ((program & AIR_ACTIVE) && (!airsupply || airsupply.air_contents.return_pressure() < 10))
+			toggleprogram(AIR_ACTIVE)
+
+		if (program & PURGE)
+			var/T = get_turf(src)
+			new /obj/effect/gibspawner/human(T)
+			spark(T, 5)
+			playsound(T, "sparks", 50, 1)
+			to_chat(victim, "<span class='warning'>You feel the machine wrestle away at your flesh with its claws! Leaving you as a husk of your former self.</span>")
```
Comment (NonQueueingMatt) [inline code/game/machinery/optableadv.dm]:
```suggestion
			to_chat(victim, "<span class='warning'>You feel the machine wrestle away at your flesh with its claws, leaving you as a husk of your former self...</span>")
```

File: code/game/machinery/optableadv.dm [hunk 39]
```diff
@@ -0,0 +1,454 @@
...
+		icon_state = victim.pulse ? "[modify_state]-lifeactive" : "[modify_state]-lifeidle"
+	else if (victim)
+		icon_state = victim.pulse ? "[modify_state]-active" : "[modify_state]-idle"
+	else
+		icon_state = "[modify_state]-idle"
+
+
+/obj/machinery/optable/lifesupport/check_table(mob/living/carbon/patient as mob)
+	if (victim)
+		usr << "\blue <B>The table is already occupied!</B>"
```
Comment (NonQueueingMatt) [inline code/game/machinery/optableadv.dm]: to_chat + notice span

File: code/game/machinery/optableadv.dm [hunk 40]
```diff
@@ -0,0 +1,454 @@
...
+	for(var/obj/item/weapon/reagent_containers/glass/G in component_parts)
+		T += G.reagents.maximum_volume
+	create_reagents(T)
+	// Transfer all reagents from the beakers to internal reagent container
+	for(var/obj/item/weapon/reagent_containers/glass/G in component_parts)
+		G.reagents.trans_to_obj(src, G.reagents.total_volume)
+
+/obj/machinery/optable/lifesupport/proc/broadcastalert(var/message as text)
+	playsound(loc, 'sound/machines/chime.ogg', 75, 1)
+	visible_message("\icon[src]<span class='warning'>The Operating Table pings: [message]</span>")
```
Comment (NonQueueingMatt) [inline code/game/machinery/optableadv.dm]:
```suggestion
	visible_message("\icon[src]<span class='warning'>[src] pings: [message]</span>")
```

File: code/game/machinery/optableadv.dm [hunk 41]
```diff
@@ -0,0 +1,454 @@
...
+					broadcastalert("ERROR: Life support systems enabled, unable to flush chemicals. Aborting!")
+					return
+				else
+					broadcastalert("Stored chemicals flushed.")
+					addtolog("Stored chemicals flushed.")
+					reagents.clear_reagents()
+
+/obj/machinery/optable/lifesupport/proc/buckle(mob/living/carbon/human/patient as mob, mob/living/carbon/human/user as mob)
+	if (!victim)
+		user << "<span class='notice'><b>No patient to buckle in!</b></span>"
```
Comment (NonQueueingMatt) [inline code/game/machinery/optableadv.dm]: to_chat

File: code/game/machinery/optableadv.dm [hunk 42]
```diff
@@ -0,0 +1,454 @@
...
+		return
+
+	if (patient == user)
+		user << "<span class='notice'><b>You cannot do this yourself!</b></span>"
```
Comment (NonQueueingMatt) [inline code/game/machinery/optableadv.dm]: to_chat

File: code/game/machinery/optableadv.dm [hunk 43]
```diff
@@ -0,0 +1,454 @@
...
+		return
+
+	if (active)
+		user << "<span class='notice'><b>You cannot do this while lifesupport is enabled!</b></span>"
```
Comment (NonQueueingMatt) [inline code/game/machinery/optableadv.dm]: to_chat

File: code/game/machinery/optableadv.dm [hunk 44]
```diff
@@ -0,0 +1,454 @@
...
+		return
+
+	if (!buckled)
+		if (patient.wear_mask)
+			user << "<span class='notice'><b>You need to remove their [patient.wear_mask] first!</b></span>"
```
Comment (NonQueueingMatt) [inline code/game/machinery/optableadv.dm]: to_chat

File: code/game/machinery/optableadv.dm [hunk 45]
```diff
@@ -0,0 +1,454 @@
...
+			return
+		else if (patient.back)
+			user << "<span class='notice'><b>You need to remove their [patient.back] first!</b></span>"
```
Comment (NonQueueingMatt) [inline code/game/machinery/optableadv.dm]: to_chat

File: code/game/machinery/optableadv.dm [hunk 46]
```diff
@@ -0,0 +1,454 @@
...
+			buckled = 0
+			patient.buckled = null
+			patient.anchored = initial(patient.anchored)
+			patient.update_canmove()
+			patient.u_equip(airmask)
+			airmask.loc = src
+			addtolog("Restraints deactivated.")
+			return
+
+/obj/machinery/optable/lifesupport/verb/bucklepatient()
```
Comment (NonQueueingMatt) [inline code/game/machinery/optableadv.dm]: How is this not generalized to optable already, why is this necessary?

File: code/game/machinery/optableadv.dm [hunk 47]
```diff
@@ -0,0 +1,454 @@
...
+				else
+					toggleprogram(AIR_ACTIVE)
+
+			active = 0
+
+			if (override)
+				broadcastalert("ERROR: Not enough chemicals to sustain lifesupport functionality! Shutting down lifesupport systems!")
+				addtolog("Emergency shutdown of life support systems.")
+			else
+				broadcastalert("Lifesupport systems disabled.")
```
Comment (NonQueueingMatt) [inline code/game/machinery/optableadv.dm]:
```suggestion
				broadcastalert("Life support systems disabled.")
```

File: code/game/mecha/equipment/weapons/mecha_ballistics.dm [hunk 48]
```diff
@@ -87,3 +87,18 @@
 	fire_sound = 'sound/effects/Explosion1.ogg'
 	projectiles = 3
 	projectile_energy_cost = 950
+
+
+/obj/item/mecha_parts/mecha_equipment/weapon/ballistic/btrgun
+	name = "M61 20mm"
```
Comment (NonQueueingMatt) [inline code/game/mecha/equipment/weapons/mecha_ballistics.dm]: This isn't consistent with all other gun names

File: code/modules/paperwork/pen.dm [hunk 49]
```diff
@@ -24,6 +24,10 @@
 	var/colour = "black"	//what colour the ink is!
 	drop_sound = 'sound/items/drop/accessory.ogg'
 
+/obj/item/weapon/pen/ispen()
+	return TRUE
+
+
```
Comment (NonQueueingMatt) [inline code/modules/paperwork/pen.dm]:
```suggestion
```

File: code/modules/research/designs.dm [hunk 50]
```diff
@@ -125,6 +125,16 @@ other types of metals and chemistry for reagents).
 	build_path = /obj/item/clothing/glasses/meson
 	sort_string = "GAAAC"
 
+
+/datum/design/item/powerdrill
+	name = "Power Drill"
```
Comment (NonQueueingMatt) [inline code/modules/research/designs.dm]:
```suggestion
	name = "power drill"
```

File: code/modules/research/designs.dm [hunk 51]
```diff
@@ -125,6 +125,16 @@ other types of metals and chemistry for reagents).
 	build_path = /obj/item/clothing/glasses/meson
 	sort_string = "GAAAC"
 
+
+/datum/design/item/powerdrill
+	name = "Power Drill"
+	desc = "A advanced drill used to make work more efficiant."
```
Comment (NonQueueingMatt) [inline code/modules/research/designs.dm]:
```suggestion
	desc = "An advanced drill designed to be faster than other drills."
```
